src/communication/gathered_vector.hpp

```diff
@@ -28,6 +30,15 @@ class gathered_vector {
         return partition_;
     }
 
+    // iterators to a distribution partition
```

Comment:
It's partially tested by Ben's com test. I should check how well, and think about how to make a better unit test from his for this.